src/main/java/com/squareup/javapoet/TypeVariableName.java

```diff
@@ -57,6 +57,18 @@ private TypeVariableName(String name, List<TypeName> bounds, List<AnnotationSpec
     return new TypeVariableName(name, bounds);
   }
 
+  public TypeVariableName withBounds(Type... bounds) {
+    return get(name, bounds);
```

Comment:
This loses any existing bounds and annotations. I think you'll want to combine just like `annotated` does.